lib/packwerk/cli.rb

```diff
@@ -9,19 +9,27 @@
 require "packwerk/formatters/progress_formatter"
 require "packwerk/inflector"
 require "packwerk/output_styles"
-require "packwerk/run_context"
+require "packwerk/node_processor_builder"
 require "packwerk/updating_deprecated_references"
 
 module Packwerk
   class Cli
     extend T::Sig
 
-    def initialize(run_context: nil, configuration: nil, out: $stdout, err_out: $stderr, style: OutputStyles::Plain)
+    def initialize(
+      node_processor_builder: nil,
+      configuration: nil,
+      out: $stdout,
+      err_out: $stderr,
+      style: OutputStyles::Plain
+    )
       @out = out
       @err_out = err_out
       @style = style
       @configuration = configuration || Configuration.from_path
-      @run_context = run_context || Packwerk::RunContext.from_configuration(@configuration)
+      reference_lister = ::Packwerk::CheckingDeprecatedReferences.new(@configuration.root_path)
```

Comment:
From my understanding of the `CheckingDeprecatedReferences` and `UpdatingDeprecatedReferences` class, both are concerned with producing a type of output for the user of packwerk. As such the initialization of the classes should happen outside the `NodeProcessorBuilder` and rather in the `Cli` class.